### PR TITLE
Add rhel9 base image

### PR DESCRIPTION
**Pull request by atheo89**
This PR related to: https://github.com/red-hat-data-services/notebooks/pull/118

The goal here is to reduce the build time for RStudio. It doesn't need to build every time the RHEL base.

The second part of this work is the modification on the OCP-CI to store the image on a registry, the related PR is here: https://github.com/openshift/release/pull/48110

### CODE REVIEW

File: base/rhel9-python-3.9/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+LABEL name="odh-notebook-base-rhel9-python-3.9" \
+      summary="Python 3.9 Red Hat Enterprise Linux 9 base image for ODH notebooks" \
+      description="Base Python 3.9 builder image based on Red Hat Enterprise Linux 9 for ODH notebooks" \
+      io.k9s.display-name="Python 3.9 RHEL9 base image for ODH notebooks" \
+      io.k9s.description="Base Python 3.9 builder image based on RHEL9 for ODH notebooks" \
+      authoritative-source-url="https://github.com/opendatahub-io/notebooks" \
+      io.openshift.build.commit.ref="main" \
+      io.openshift.build.source-location="https://github.com/opendatahub-io/notebooks/tree/main/base/rhel9-python-3.9" \
+      io.openshift.build.image="quay.io/opendatahub/workbench-images:base-rhel9-python-3.9"
```
Comment (harshad16) [inline base/rhel9-python-3.9/Dockerfile:11]: As this is exclusive to the downstream repository, best to have this configured exclusively for this repo.

```suggestion
LABEL name="rhoai-notebook-base-rhel9-python-3.9" \
      summary="Python 3.9 Red Hat Enterprise Linux 9 base image for RHOAI notebooks" \
      description="Base Python 3.9 builder image based on Red Hat Enterprise Linux 9 for RHOAI notebooks" \
      io.k9s.display-name="Python 3.9 RHEL9 base image for RHOAI notebooks" \
      io.k9s.description="Base Python 3.9 builder image based on RHEL9 for RHOAI notebooks" \
      authoritative-source-url="https://github.com/red-hat-data-services/notebooks" \
      io.openshift.build.commit.ref="main" \
      io.openshift.build.source-location="https://github.com/red-hat-data-services/notebooks/tree/main/base/rhel9-python-3.9" \
      io.openshift.build.image="quay.io/modh/odh-base-rhel9"
```

File: manifests/base/base-rhel9-imagestream.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (harshad16) [inline manifests/base/base-rhel9-imagestream.yaml:1]: we dont require manifest files for this ,
please revert the imagestream change for base-rhel9

remove the changes under  manifests/base

**Review by harshad16 — CHANGES_REQUESTED** [review]
thanks for the PR
shared some reviews.

File: base/rhel9-python-3.9/Dockerfile (hunk 1 above)
Comment (harshad16) [reply base/rhel9-python-3.9/Dockerfile:11]: Maybe we should introduce this file structure in upstream.
Not build it, just the file structure.
so that we can follow the same pattern.
WDYT?

Comment (atheo89) [reply base/rhel9-python-3.9/Dockerfile:11]: I think it would be a good practice.

File: manifests/base/base-rhel9-imagestream.yaml (hunk 2 above)
Comment (atheo89) [reply manifests/base/base-rhel9-imagestream.yaml]: removed

File: manifests/base/params.yaml [hunk 3]
```diff
@@ -75,4 +75,5 @@ varReference:
   - path: spec/tags[]/from/name
     kind: ImageStream
     apiGroup: image.openshift.io/v1
-    name: odh-codeserver-notebook-n
\ No newline at end of file
+    name: odh-codeserver-notebook-n
```
Comment (harshad16) [inline manifests/base/params.yaml:78]: please revert this change, we dont require this.

**Review by harshad16 — CHANGES_REQUESTED** [review]
all looks good
one small change needed.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve 
:+1:
